src/bp-messages/classes/class-bp-messages-message.php

```diff
@@ -323,13 +323,15 @@ public static function is_user_sender( $user_id, $message_id ) {
 	 * @return int|null The ID of the sender if found, otherwise null.
 	 */
 	public static function get_message_sender( $message_id ) {
-		global $wpdb;
-
-		$bp = buddypress();
-
-		$query = $wpdb->get_var( $wpdb->prepare( "SELECT sender_id FROM {$bp->messages->table_name_messages} WHERE id = %d", $message_id ) );
+		
+		$query = BP_Messages_Message::get(
+			array(
+				'fields'  => 'sender_ids',
+				'include' => array( $message_id ),
+			)
+		);
 
-		return is_numeric( $query ) ? (int) $query : $query;
+		return ( isset( $query['messages'][0] ) && is_numeric( $query ) ) ? (int) $query['messages'][0] : null;
```

Comment:
@janakkaneriya007 `is_numeric( $query )` condition wrong i think you should check `is_numeric( $query['messages'][0] )`